.circleci/config.yml

```diff
@@ -7,11 +7,9 @@ jobs:
         steps:
             - checkout
 
-            - save_cache:
-                key: dependency-cache
-                paths:
-                    - ~/.composer
-                    - ~/Library/Caches/Homebrew
+            - restore_cache:
+                keys:
+                    - source-v1-{{ checksum "composer.json" }}
```

Comment:
@keradus what about this key?

Reply:
no hard opinion on this one

```
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer λ composer info | wc -l
64
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer λ git diff 2.12..3.0 composer.json
diff --git a/composer.json b/composer.json
index 5df4718f..b42e19aa 100644
--- a/composer.json
+++ b/composer.json
@@ -20,7 +20,7 @@
         "composer/semver": "^1.4",
         "composer/xdebug-handler": "^1.2",
         "doctrine/annotations": "^1.2",
-        "php-cs-fixer/diff": "^1.3",
+        "php-cs-fixer/diff": "^2.0",
         "symfony/console": "^3.4.17 || ^4.1.6",
         "symfony/event-dispatcher": "^3.0 || ^4.0",
         "symfony/filesystem": "^3.0 || ^4.0",
```

if we have 64 package installed, do we really want to not use cache of them, if **one** package is different ? installing 63 packages from cache + one fresh vs 64 fresh...

Reply:
Yes, and then it will cache them and reuse again - bear in mind that CircleCI will not save the cache when there already is anything saved under the key - take a look [here](https://circleci.com/gh/FriendsOfPHP/PHP-CS-Fixer/5470)  in "Save Cache" action - 1 month old cache, impossible to track what is there.

Reply:
oh damn it !
ok, then your proposal of using checksum makes perfect sense !

why `source-v1` prefix ? what does it stand for ?

Reply:
I have no idea - it's from the [docs](https://circleci.com/docs/2.0/caching/#source-caching) - never thought about it. Example seems to cache source and we would cache dependencies and Homebrew - what name would you propose? maybe simply `cache`?